aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java

```diff
@@ -144,6 +144,32 @@ public void testDefaultWarehouseLocationDbUri() {
     Assert.assertEquals("s3://bucket2/db/table", location);
   }
 
+  @Test
+  public void testDefaultWarehouseLocationCustomCatalogId() {
+    GlueCatalog catalogWithCustomCatalogId = new GlueCatalog();
+    String catalogId = "myCatalogId";
+    AwsProperties awsProperties = new AwsProperties();
+    awsProperties.setGlueCatalogId(catalogId);
+    catalogWithCustomCatalogId.initialize(
+        CATALOG_NAME,
+        WAREHOUSE_PATH + "/",
+        awsProperties,
+        glue,
+        LockManagers.defaultLockManager(),
+        null,
+        ImmutableMap.of());
+
+    Mockito.doReturn(
+            GetDatabaseResponse.builder()
+                .database(Database.builder().name("db").locationUri("s3://bucket2/db").build())
+                .build())
+        .when(glue)
+        .getDatabase(Mockito.any(GetDatabaseRequest.class));
+    catalogWithCustomCatalogId.defaultWarehouseLocation(TableIdentifier.of("db", "table"));
+    Mockito.verify(glue)
+        .getDatabase(Mockito.argThat((GetDatabaseRequest req) -> req.catalogId() == catalogId));
```

Comment:
Since catalogId is a string can we do the equality check via .equals?

```
Mockito.argThat((GetDatabaseRequest req) -> catalogId.equals(req.catalogId()))
```